packages/angular/projects/clr-angular/src/data/datagrid/providers/selection.ts

```diff
@@ -237,8 +238,10 @@ export class Selection<T = any> {
 
   public updateCurrent(value: T[], emit: boolean) {
     this._current = value;
-    if (emit) {
+    if (emit && !this.debounce) {
       this.emitChange();
+      this.debounce = true;
+      setTimeout(() => (this.debounce = false));
```

Comment:
This pattern raises a flag for me, what exactly is happening that requires this? I can't tell from the code review what this solves.

Reply:
I added this because otherwise the change was emitted with every item that was added to the selection (within the same change detection cycle). This made the following test fail: https://github.com/vmware/clarity/blob/9c5523481c1ed574a3b6d57de99df2dc6d8637a2/packages/angular/projects/clr-angular/src/data/datagrid/providers/selection.spec.ts#L182 

I saw that this debounce was also in previous implementations of Selection.

Reply:
We're usually unwilling to use setTimeout, unless unavoidable, as it is patched from the Angular framework and forces a change detection cycle on every call.
The possible alternatives we've been using are either use an observable with debounceTime(0) (as in popover-content.ts), or call the setTimeout in an ngZone's runOutsideAngular, as done in datagrid-action-overflow.ts
The first may be more suitable in this case, as it's also about debouncing "nearly simultaneous" events, while the second we usually use with focus management, which really isn't covered well in Angular.

Reply:
I tried implementing debounceTime(0), but that caused another test to fail: https://github.com/vmware/clarity/blob/ebb6f552485fe82d1f103f568548cf32c3d7ca09/packages/angular/projects/clr-angular/src/data/datagrid/providers/selection.spec.ts#L214

It seems to emit too late for the test to check the amount of changes. But all other tests do succeed. Is this an issue with the test?